### ISSUE TITLE
font.getGlyphID() returns None for missing glyphs unless they start with `glyph`

### ISSUE THREAD

**Issue by behdad** [fonttools/fonttools#3030]
Don't know how this went unnoticed but I found this while debugging the cryptic error message in https://github.com/fonttools/fonttools/issues/3028

Basically this function can fall off and return None by default:
```
    def getGlyphID(self, glyphName): 
        """Returns the ID of the glyph with the given name.""" 
        try: 
            print(glyphName) 
            return self.getReverseGlyphMap()[glyphName] 
        except KeyError: 
            if glyphName[:5] == "glyph": 
                try: 
                    return int(glyphName[5:]) 
                except (NameError, ValueError): 
                    raise KeyError(glyphName) 
```

It should `raise KeyError(glyphName)` instead.

**Comment by khaledhosny** [comment]
Is this code related to virtual glyph IDs? There was also a probably related comment in `getGlyphName()` that was removed in https://github.com/fonttools/fonttools/commit/322a7b2ead069404e6fb278ab0a904999ab78bb9:
https://github.com/fonttools/fonttools/blob/49b6004040902e8b13f1aa7327769872bafa12a9/Lib/fontTools/ttLib/ttFont.py#L521-L528

**Comment by anthrotype** [comment]
yes, it looks like that one is the commit which introduced the bug.

> It should raise KeyError(glyphName) instead.

sounds good, I shall do that
